tools/install_demo_configuration.bat

```diff
@@ -319,6 +321,47 @@ echo plugins.security.restapi.roles_enabled: ["all_access", "security_rest_api_a
 echo plugins.security.system_indices.enabled: true >> "%OPENSEARCH_CONF_FILE%"
 echo plugins.security.system_indices.indices: [".plugins-ml-config", ".plugins-ml-connector", ".plugins-ml-model-group", ".plugins-ml-model", ".plugins-ml-task", ".plugins-ml-conversation-meta", ".plugins-ml-conversation-interactions", ".opendistro-alerting-config", ".opendistro-alerting-alert*", ".opendistro-anomaly-results*", ".opendistro-anomaly-detector*", ".opendistro-anomaly-checkpoints", ".opendistro-anomaly-detection-state", ".opendistro-reports-*", ".opensearch-notifications-*", ".opensearch-notebooks", ".opensearch-observability", ".ql-datasources", ".opendistro-asynchronous-search-response*", ".replication-metadata-store", ".opensearch-knn-models", ".geospatial-ip2geo-data*", ".opendistro-job-scheduler-lock"] >> "%OPENSEARCH_CONF_FILE%"
 
+setlocal enabledelayedexpansion
+:: Read the admin password from the file or use the initialAdminPassword if set
+for /f %%a in ('type "%ADMIN_PASSWORD_FILE%"') do set "ADMIN_PASSWORD=%%a"
```

Comment:
I'd recommend making the order shown in the file represents the order of importance of the source.  In this case, first we would check if `initialAdminPassword` env variable had a value, then, only if it didn't, we would check the file.  Then we'd have a final `if not defined ADMIN_PASSWORD` check that would throw the error message and exit the script.

Reply:
This avoids checking for the password file to exist, reading it, and then making it the password only to override it later

Reply:
Will do